### PR TITLE
[Enhancement](Load) stream tvf support json

### LINKED ISSUE #23678: [Enhancement] Improve the functionality of http_stream tvf
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

We still need to do the following functions now
1. Support file import in CSV header format
2. Support for importing JSON file format
3. Support for two phase commit
4. If the above functions are complete, it is necessary to convert the original stream_load format into http_stream mode for import

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zzzzzzzs**
## Proposed changes

Issue Number: close https://github.com/apache/doris/issues/23678

## Further comments
stream tvf support json

```
[{"id":1, "name":"ftw", "age":18}]
[{"id":2, "name":"xxx", "age":17}]
[{"id":3, "name":"yyy", "age":19}]
```

example:
``` curl
curl -v --location-trusted -u root: -H "sql: insert into test.t1(c1, c2) select id, name from http_stream(\"format\" = \"json\", \"strip_outer_array\" = \"true\", \"read_json_by_line\" = \"true\")" -T /root/json_file.json http://127.0.0.1:8030/api/_http_stream

```

### CODE REVIEW

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.29 seconds
 stream load tsv:          538 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162058540 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/ExternalFileTableValuedFunction.java [hunk 1]
```diff
@@ -227,8 +227,7 @@ protected void parseProperties(Map<String, String> validParams) throws AnalysisE
         }
 
         // TODO Support is needed in the future
-        if (getTFileType() == TFileType.FILE_STREAM && (formatString.equals("json")
-                || formatString.equals("csv_with_names")
+        if (getTFileType() == TFileType.FILE_STREAM && (formatString.equals("csv_with_names")
                 || formatString.equals("csv_with_names_and_types")
                 || formatString.equals("parquet")
                 || formatString.equals("avro")
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/ExternalFileTableValuedFunction.java:233]: throw new AnalysisException("current http_stream does not yet support **json**, parquet and orc");

File: regression-test/suites/load_p0/http_stream/test_http_stream_json.groovy [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                }
+                log.info("http_stream result: ${result}".toString())
+                def json = parseJson(result)
+                assertEquals("success", json.Status.toLowerCase())
+                assertEquals(12, json.NumberTotalRows)
+                assertEquals(0, json.NumberFilteredRows)
+            }
+        }
+
+        qt_sql1 "select id, city, code from ${tableName1}"
```
Comment (yiguolei) [inline regression-test/suites/load_p0/http_stream/test_http_stream_json.groovy:61]: add order by

File: regression-test/suites/load_p0/http_stream/test_http_stream_json.groovy [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                }
+                log.info("http_stream result: ${result}".toString())
+                def json = parseJson(result)
+                assertEquals("success", json.Status.toLowerCase())
+                assertEquals(10, json.NumberTotalRows)
+                assertEquals(0, json.NumberFilteredRows)
+            }
+        }
+
+        qt_sql2 "select id, city, code from ${tableName2}"
```
Comment (yiguolei) [inline regression-test/suites/load_p0/http_stream/test_http_stream_json.groovy:104]: add order by

File: regression-test/suites/load_p0/http_stream/test_http_stream_json.groovy [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+                }
+                log.info("http_stream result: ${result}".toString())
+                def json = parseJson(result)
+                assertEquals("success", json.Status.toLowerCase())
+                assertEquals(12, json.NumberTotalRows)
+                assertEquals(0, json.NumberFilteredRows)
+            }
+        }
+
+        qt_sql4 "select id, code from ${tableName4}"
```
Comment (yiguolei) [inline regression-test/suites/load_p0/http_stream/test_http_stream_json.groovy:192]: add order by

File: regression-test/suites/load_p0/http_stream/test_http_stream_json.groovy [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+                }
+                log.info("http_stream result: ${result}".toString())
+                def json = parseJson(result)
+                assertEquals("success", json.Status.toLowerCase())
+                assertEquals(5, json.NumberTotalRows)
+                assertEquals(0, json.NumberFilteredRows)
+            }
+        }
+
+        qt_sql3 "select id, city, code from ${tableName3}"
```
Comment (yiguolei) [inline regression-test/suites/load_p0/http_stream/test_http_stream_json.groovy:148]: add order by

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.3 seconds
 stream load tsv:          539 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17161868837 Bytes

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**